core/src/main/java/com/sequenceiq/cloudbreak/repository/StackRepository.java

```diff
@@ -41,50 +41,82 @@
             + "LEFT JOIN FETCH ig.securityGroup LEFT JOIN FETCH ig.instanceGroupNetwork WHERE s.resourceCrn= :crn " +
-            "AND s.workspace.id= :workspaceId AND " + SHOW_TERMINATED_CLUSTERS_IF_REQUESTED)
+            "AND s.workspace.id= :workspaceId AND s.detached = null AND " + SHOW_TERMINATED_CLUSTERS_IF_REQUESTED)
     Optional<Stack> findByCrnAndWorkspaceIdWithLists(@Param("crn") String crn, @Param("workspaceId") Long workspaceId,
             @Param("showTerminated") Boolean showTerminated, @Param("terminatedAfter") Long terminatedAfter);
 
     @Query("SELECT s FROM Stack s LEFT JOIN FETCH s.instanceGroups ig LEFT JOIN FETCH ig.instanceMetaData LEFT JOIN FETCH ig.template "
             + "LEFT JOIN FETCH ig.securityGroup LEFT JOIN FETCH ig.instanceGroupNetwork WHERE s.resourceCrn= :crn " +
-            "AND s.workspace.id= :workspaceId AND s.type = :type AND " +
+            "AND s.workspace.id= :workspaceId AND s.detached is not null AND " + SHOW_TERMINATED_CLUSTERS_IF_REQUESTED)
+    Optional<Stack> findByCrnAndWorkspaceIdWithListsAndDetached(@Param("crn") String crn, @Param("workspaceId") Long workspaceId,
+            @Param("showTerminated") Boolean showTerminated, @Param("terminatedAfter") Long terminatedAfter);
+
+    @Query("SELECT s FROM Stack s LEFT JOIN FETCH s.instanceGroups ig LEFT JOIN FETCH ig.instanceMetaData LEFT JOIN FETCH ig.template "
+            + "LEFT JOIN FETCH ig.securityGroup LEFT JOIN FETCH ig.instanceGroupNetwork WHERE s.resourceCrn= :crn " +
+            "AND s.workspace.id= :workspaceId AND s.type = :type AND s.detached = null AND " +
             SHOW_TERMINATED_CLUSTERS_IF_REQUESTED)
     Optional<Stack> findByCrnAndWorkspaceIdWithLists(@Param("crn") String crn, @Param("type") StackType type, @Param("workspaceId") Long workspaceId,
             @Param("showTerminated") Boolean showTerminated, @Param("terminatedAfter") Long terminatedAfter);
 
+    @Query("SELECT s FROM Stack s LEFT JOIN FETCH s.instanceGroups ig LEFT JOIN FETCH ig.instanceMetaData LEFT JOIN FETCH ig.template "
+            + "LEFT JOIN FETCH ig.securityGroup LEFT JOIN FETCH ig.instanceGroupNetwork WHERE s.resourceCrn= :crn " +
+            "AND s.workspace.id= :workspaceId AND s.type = :type AND s.detached is not null AND " +
+            SHOW_TERMINATED_CLUSTERS_IF_REQUESTED)
+    Optional<Stack> findByCrnAndWorkspaceIdWithListsAndDetached(@Param("crn") String crn, @Param("type") StackType type, @Param("workspaceId") Long workspaceId,
+            @Param("showTerminated") Boolean showTerminated, @Param("terminatedAfter") Long terminatedAfter);
+
     @Query("SELECT s FROM Stack s LEFT JOIN FETCH s.instanceGroups ig LEFT JOIN FETCH ig.instanceMetaData LEFT JOIN FETCH ig.template " +
             "LEFT JOIN FETCH ig.securityGroup LEFT JOIN FETCH ig.instanceGroupNetwork WHERE s.id= :id AND (s.type is not 'TEMPLATE' OR s.type is null)")
```

Comment:
Shouldn't this be changed as well?

Reply:
I did not want to change the query that's already there. I re-used the same and extended it to include detached check.